scripts/list_provider_vms.py

```diff
@@ -0,0 +1,145 @@
+import argparse
+from tabulate import tabulate
+from multiprocessing import Process, Queue
+from multiprocessing.queues import Empty
+
+from utils.conf import cfme_data
+from utils.path import log_path
+from utils.providers import get_mgmt, list_provider_keys
+
+
+# Constant for report
+NULL = '--'
+
+
+def parse_cmd_line():
+    """
+    Specify and parse arguments
+    :return: args, kwargs, the usual
+    """
+    parser = argparse.ArgumentParser(argument_default=None)
+    parser.add_argument('--outfile',
+                        default=log_path.join('list_provider_vms.log').strpath,
+                        dest='outfile')
+    parser.add_argument('--tag',
+                        default=None,
+                        dest='tag',
+                        action='append',
+                        help='A provider tag to match a group of providers instead of all '
+                             'providers from cfme_data. Can be used multiple times')
+    parser.add_argument('provider',
+                        default=None,
+                        nargs='*',
+                        help='Provider keys, can be user multiple times. If none are given '
+                             'the script will use all providers from cfme_data or match tags')
+
+    args = parser.parse_args()
+    return args
+
+
+def process_tags(provider_keys, tags=[]):
+    """
+    Process the tags provided on command line to build a list of provider keys that match
+    :param tags: list of tags to match against cfme_data
+    :param provider_keys list of provider_keys to append to
+    :return: list or provider keys matching tags
+    """
+    # Check for tags first, build list of provider_keys from it
+    if tags:
+        all_provider_keys = list_provider_keys()
+        for key in all_provider_keys:
+            # need to check tags list against yaml tags list for intersection of a single tag
+            yaml_tags = cfme_data['management_systems'][key]['tags']
+            if any(tag in tags for tag in yaml_tags):
+                print('Matched tag from {} on provider {}:tags:{}'.format(tags, key, yaml_tags))
+                provider_keys.append(key)
+
+
+def list_vms(provider_key, output_queue):
+    """
+    List all the vms/instances on the given provider key
+    Build list of lists with basic vm info: [[provider, vm, status, age, type], [etc]]
+    :param provider_key: string provider key
+    :param output_queue: a multiprocessing.Queue object to add results to
+    :return: list of lists of vms and basic statistics
+    """
+    output_list = []
+
+    print('Listing VMS on provider {}'.format(provider_key))
+    provider = get_mgmt(provider_key)
+    try:
```

Comment:
ah, yeah good call. I shoehorned in the NotImplementedCatch at the end and its ugly as-is.